Source/WarpX.cpp

```diff
@@ -257,6 +257,56 @@ WarpX::WarpX ()
         default: amrex::Abort("unknown load balance type");
     }
 
+    // Set default values for particle and cell weights for costs update;
+    // Default values listed here for the case AMREX_USE_GPU are determined
+    // from single-GPU tests on Summit.
+#ifdef AMREX_USE_GPU
+    if (costs_heuristic_cells_wt==-1 && costs_heuristic_particles_wt==-1
+        && WarpX::load_balance_costs_update_algo==LoadBalanceCostsUpdateAlgo::Heuristic)
+    {
+#ifdef WARPX_USE_PSATD
+        switch (WarpX::nox)
+        {
+            case 1:
+                costs_heuristic_cells_wt = 0.575;
+                costs_heuristic_particles_wt = 0.425;
+                break;
+            case 2:
+                costs_heuristic_cells_wt = 0.405;
+                costs_heuristic_particles_wt = 0.595;
+                break;
+            case 3:
+                costs_heuristic_cells_wt = 0.250;
+                costs_heuristic_particles_wt = 0.750;
+                break;
+        }
+#else // FDTD
+        switch (WarpX::nox)
+        {
+            case 1:
+                costs_heuristic_cells_wt = 0.401;
+                costs_heuristic_particles_wt = 0.599;
+                break;
+            case 2:
+                costs_heuristic_cells_wt = 0.268;
+                costs_heuristic_particles_wt = 0.732;
+                break;
+            case 3:
+                costs_heuristic_cells_wt = 0.145;
+                costs_heuristic_particles_wt = 0.855;
+                break;
+        }
+#endif // WARPX_USE_PSATD
+    }
+#else // CPU
```

Comment:
That look a bit weird. We have
```
#ifdef AMREX_USE_GPU
    if (costs_heuristic_cells_wt==-1 && costs_heuristic_particles_wt==-1...)
    {
 #ifdef WARPX_USE_PSATD
#endif // WARPX_USE_PSATD
     }
 #else // CPU
    if (costs_heuristic_cells_wt==-1 && costs_heuristic_particles_wt==-1...)
    {
    }
#endif 
```
Why not
```
    if (costs_heuristic_cells_wt==-1 && costs_heuristic_particles_wt==-1...)
    {
#ifdef AMREX_USE_GPU
 #ifdef WARPX_USE_PSATD
#endif // WARPX_USE_PSATD
 #else // CPU
#endif 
}
```
?

Reply:
Thanks, this clarifies the logic.  It is changed.